File: backend/hackathon-backend/check_db.py
#!/usr/bin/env python3
"""
Script to check database schema and data persistence
"""

import sqlite3
import os

def check_database():
    """Check database schema and data"""
    db_path = "hackathon.db"
    
    if not os.path.exists(db_path):
        print(f"❌ Database file {db_path} does not exist")
        return
    
    print(f"✅ Database file {db_path} exists")
    
    try:
        conn = sqlite3.connect(db_path)
        cursor = conn.cursor()
        
        # Check tables
        cursor.execute("SELECT name FROM sqlite_master WHERE type='table'")
        tables = cursor.fetchall()
        print(f"\n📊 Tables in database:")
        for table in tables:
            print(f"  - {table[0]}")
        
        # Check if alembic_version table exists (indicates migrations applied)
        cursor.execute("SELECT * FROM sqlite_master WHERE name='alembic_version'")
        alembic_table = cursor.fetchone()
        if alembic_table:
            print(f"\n✅ Alembic migrations have been applied")
            cursor.execute("SELECT * FROM alembic_version")
            version = cursor.fetchone()
            print(f"   Current migration version: {version[0] if version else 'Unknown'}")
        else:
            print(f"\n⚠️  Alembic version table not found - migrations may not be applied")
        
        # Check for key tables from models
        key_tables = ['users', 'deliverables', 'sprints', 'signoffs', 'user_profiles', 'user_settings']
        print(f"\n🔍 Checking key application tables:")
        
        for table in key_tables:
            cursor.execute(f"SELECT name FROM sqlite_master WHERE type='table' AND name='{table}'")
            exists = cursor.fetchone()
            if exists:
                print(f"  ✅ {table} table exists")
                # Count rows
                cursor.execute(f"SELECT COUNT(*) FROM {table}")
                count = cursor.fetchone()[0]
                print(f"     Rows: {count}")
            else:
                print(f"  ❌ {table} table does not exist")
        
        # Show registered users
        print(f"\n👥 Registered users:")
        cursor.execute("SELECT id, email, first_name, last_name FROM users")
        users = cursor.fetchall()
        for user in users:
            print(f"  ID: {user[0]}, Email: {user[1]}, Name: {user[2]} {user[3]}")
        
        conn.close()
        
    except sqlite3.Error as e:
        print(f"❌ Database error: {e}")

if __name__ == "__main__":
    check_database()

File: backend/check-db.js
const { Pool } = require('pg');
const dbConfig = require('./database-config');

async function checkDatabase() {
  const pool = new Pool(dbConfig);
  let client;
  
  try {
    client = await pool.connect();
    
    // Check users table columns
    const result = await client.query(`
      SELECT column_name, data_type, is_nullable 
      FROM information_schema.columns 
      WHERE table_name = 'users'
      ORDER BY ordinal_position
    `);
    
    console.log('Users table columns:');
    result.rows.forEach(row => {
      console.log(`  - ${row.column_name} (${row.data_type}, ${row.is_nullable === 'YES' ? 'nullable' : 'not null'})`);
    });
    
    // Check if admin user exists
    const adminResult = await client.query('SELECT email, first_name, last_name, role FROM users WHERE email = $1', ['admin@flowspace.com']);
    console.log('\nAdmin user:');
    if (adminResult.rows.length > 0) {
      console.log(`  - Email: ${adminResult.rows[0].email}`);
      console.log(`  - Name: ${adminResult.rows[0].first_name} ${adminResult.rows[0].last_name}`);
      console.log(`  - Role: ${adminResult.rows[0].role}`);
    } else {
      console.log('  - Not found');
    }
    
  } catch (error) {
    console.error('Error checking database:', error.message);
  } finally {
    if (client) await client.release();
    await pool.end();
  }
}

checkDatabase();